src/ips.rs

```diff
@@ -0,0 +1,525 @@
+// Intelligent Peer Sharing (IPS) module
+// Selection is based on the "beauty" contest of the nodes - each node is evaluated based on its
+// degree, betweenness, closeness and eigenvector centrality. Then, if requested ranking is
+// updated with location factor. Each factor has its own weight that is used to determine
+// factor's importance to the calculation of the final ranking. That gives possibility
+// to test different approaches to the selection of the peers, without re-compiling the code.
+// Weights are defined in the configuration file.
+// When the ranking is calculated, the peers are selected based on the ranking. The number of
+// peers can be changed and it is defined in the configuration file. Algorithm is constructed
+// as step by step process, so it is easy to add new steps or change the order of the steps.
+// Especially, there could be a need to add some modifiers to the ranking.
+
+use crate::config::IPSConfiguration;
```

Comment:
Random note on imports: I quite like having them formatted by `rustfmt`, I usually use e.g.: https://github.com/niklaslong/spectre/blob/main/.rustfmt.toml. (note, one of those might require the nightly formatter).

Reply:
Formatter added.